react/features/base/react/components/native/PagedList.js

```diff
@@ -0,0 +1,287 @@
+     */
+    pages: Array<{
+        component: ?Object,
+        icon: string | number,
+        title: string
+    }>
+};
+
+/**
+ * The type of the React {@code Component} state of {@link PagedList}.
+ */
+type State = {
+
+    /**
+     * The currently selected page.
+     */
+    pageIndex: number
+};
+
+/**
+ * A component that renders a paged list.
+ *
+ * @extends PagedList
+ */
+class PagedList extends Component<Props, State> {
+
+    /**
+     * Initializes a new {@code PagedList} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            pageIndex: this._validatePageIndex(props.defaultPage)
+        };
+
+        // Bind event handlers so they are only bound once per instance.
+        this._maybeRefreshSelectedPage
+            = this._maybeRefreshSelectedPage.bind(this);
+    }
+
+    /**
+     * Renders the component.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled } = this.props;
+        const pages = this.props.pages.filter(({ component }) => component);
+
+        return (
+            <View
+                style = { [
+                    styles.pagedListContainer,
+                    disabled ? styles.pagedListContainerDisabled : null
+                ] }>
+                {
+                    pages.length > 1
+                        ? this._renderPagedList(disabled)
+                        : React.createElement(
+
+                            // $FlowExpectedError
+                            /* type */ pages[0].component,
+                            /* props */ {
+                                disabled,
+                                style: styles.pagedList
+                            })
+                }
+            </View>
+        );
+    }
+
+    /**
+     * Constructs the style of an indicator.
+     *
+     * @param {number} indicatorIndex - The index of the indicator.
+     * @private
+     * @returns {Object}
+     */
+    _getIndicatorStyle(indicatorIndex) {
+        if (this.state.pageIndex === indicatorIndex) {
+            return styles.pageIndicatorActive;
+        }
+
+        return null;
+    }
+
+    _maybeRefreshSelectedPage: ?boolean => void;
+
+    /**
+     * Components that this PagedList displays may have a refresh function to
+     * refresh its content when displayed (or based on custom logic). This
+     * function invokes this logic if it's present.
+     *
+     * @private
+     * @param {boolean} isInteractive - If true this refresh was caused by
+     * direct user interaction, false otherwise.
+     * @returns {void}
+     */
+    _maybeRefreshSelectedPage(isInteractive: boolean = true) {
+        const selectedPage = this.props.pages[this.state.pageIndex];
+        let component;
+
+        if (selectedPage && (component = selectedPage.component)) {
+            const { refresh } = component;
+
+            typeof refresh === 'function'
```

Comment:
we should now be able to call the function unconditionally, since it's defined int eh abstract class